src/ProxyClient/MultiplexerClient.php

```diff
@@ -126,18 +160,51 @@ public function refresh($url, array $headers = [])
     }
 
     /**
-     * Helper function to invoke the given $method on all available ProxyClients implementing the given $interface.
+     * Invoke the given $method on all available ProxyClients implementing the
+     * given $interface.
      *
      * @param string $interface The FQN of the interface
      * @param string $method    The method to invoke
      * @param array  $arguments The arguments to be passed to the method
      */
     private function invoke($interface, $method, array $arguments)
     {
-        foreach ($this->proxyClients as $proxyClient) {
-            if (is_subclass_of($proxyClient, $interface)) {
-                call_user_func_array([$proxyClient, $method], $arguments);
-            }
+        foreach ($this->getProxyClients($interface) as $proxyClient) {
+            call_user_func_array([$proxyClient, $method], $arguments);
         }
     }
+
+    /**
+     * Invoke the given $method on the first available ProxyClient implementing
+     * the given $interface.
+     *
+     * @param string $interface The FQN of the interface
+     * @param string $method    The method to invoke
+     * @param array  $arguments The arguments to be passed to the method
+     *
+     * @return mixed Return value of ProxyClient method call
+     */
+    private function invokeFirst($interface, $method, array $arguments)
+    {
+        foreach ($this->getProxyClients($interface) as $proxyClient) {
+            return call_user_func_array([$proxyClient, $method], $arguments);
+        }
```

Comment:
maybe you can review and merge #346 and then add a bit about exceptions for tag operations in this PR?